app/models/manager_refresh/graph.rb

```diff
@@ -10,6 +10,38 @@ def initialize(nodes)
       construct_graph!(@nodes)
     end
 
+    def to_graphviz(layers: nil)
+      node_names = {}
+      # Want readable ids, but can't have duplicates, so append numbers.
+      # Try to use shorter .name method that InventoryCollection has.
+      nodes.group_by {|n| n.name.to_s rescue n.to_s }.each do |base_name, ns|
```

Comment:
> n.name.to_s rescue n.to_s

Not crazy about this rescue, how about a `node_friendly_name` helper which checks if `n.kind_of? InventoryCollection` or `n.respond_to? :name` ?